patches/api/0114-Add-EntityKnockbackByEntityEvent-and-EntityPushedByE.patch

```diff
@@ -28,11 +29,20 @@ index 0000000000000000000000000000000000000000..e0ba692c9b107f2b042a9c06549185e1
 +public class EntityKnockbackByEntityEvent extends EntityPushedByEntityAttackEvent {
 +
 +    private final float knockbackStrength;
++    private final KnockbackCause cause;
 +
 +    @ApiStatus.Internal
 +    public EntityKnockbackByEntityEvent(@NotNull LivingEntity entity, @NotNull Entity hitBy, float knockbackStrength, @NotNull Vector acceleration) {
 +        super(entity, hitBy, acceleration);
 +        this.knockbackStrength = knockbackStrength;
++        this.cause = KnockbackCause.UNKNOWN;
++    }
++
++    @ApiStatus.Internal
++    public EntityKnockbackByEntityEvent(@NotNull LivingEntity entity, @NotNull Entity hitBy, @NotNull final KnockbackCause cause, float knockbackStrength, @NotNull Vector acceleration) {
```

Comment:
I'm not sure if the compatibility is needed here now that all the constructors are internal anyway but if not the old constructor should be marked as deprecated.